test/subroutine/association_test.rb

```diff
@@ -335,5 +335,15 @@ def test_groups_are_preserved_to_association_components
       assert_equal({}, op.without_info_params)
     end
 
+    def test_find_by_is_used_if_raise_on_miss_is_false
```

Comment:
do we have coverage of the opposite? i looked quick and didn't see anything ad-hoc
mainly to make sure that this change (or future) don't break that behavior

Reply:
There are other tests in here that use set expectations on `find_by!()` usage